### DISCUSSION TITLE
Deployment

### DISCUSSION THREAD

**Discussion by 20May1997** [SYSTRAN/faster-whisper#655] [General]
Can anybody tell me how can i run deplay this model using flask api I am fetching the audio file from a url path. but unable to handle the audio file

**Comment by trungkienbkhn** [comment]
@20May1997, hello. You can try sending your audio file in the flask request files, then call the model to execute this file.
```python
@app.route("/transcribe", methods=["POST"])
def transcribe():
    input_file = flask.request.files.get("input")
    try:
        segments, info = model.transcribe(input_file)
        segments = list(segments)
        response = {
            "transcription": "".join(segment.text for segment in segments),
            "language": info.language,
            "duration": info.duration,
        }
        return response
    except Exception as e:
        return message("Internal error: %s" % e, 500)
``` 
Hope it will be useful for you.

**Reply by 20May1997** [reply]
Thank you so much for the help .. Actually I was given a url path which is the location of the audio file . Have you implemented by yourself . If possible please provide me the git repo I don't know why but I am still getting a error while trying to display the output on a website page

**Reply by trungkienbkhn** [reply]
Not yet. I haven't tried with audio url before. Can you give an url example ?
Also, I think we should download/get the audio data from the url, convert to npdarray and then send it to the model.

**Reply by 20May1997** [reply]
So basically the https://demo/recordings/demo.mp3

The url path looks like this if possible please provide me the git repo or proper app.py to do it because I am struggling from past 4 days 😅😅

**Reply by trungkienbkhn** [reply]
I think you can use the ffmpeg lib to get the audio data from url path. I will try this.

**Reply by 20May1997** [reply]
So basically when I try to show the results in html page as well i doesn't give any output. Just the blank space

**Reply by 20May1997** [reply]
Sure please let me know if it works and please share the code as well

**Reply by trungkienbkhn** [reply]
```python
import io
import faster_whisper as fw
import requests
import flask
from flask import Flask, jsonify

app = Flask(__name__)
model = fw.WhisperModel("tiny")


def load_audio_from_url(url):
    response = requests.get(url, stream=True)

    if response.status_code == 200:
        audio_data = io.BytesIO(response.content)
        return audio_data
    else:
        print(f"Failed to download audio")
        return None


@app.route('/transcribe', methods=['GET'])
def hello():
    # audio_url = flask.request.json.get('url')
    audio_url = "https://6a63fca904fd268f15f7-d5770ffdd579eb31eaa89faeffc55fe7.ssl.cf1.rackcdn.com/Audio_zone-A_special_restaurant.mp3"
    data = load_audio_from_url(audio_url)
    segments, info = model.transcribe(data, language='en')
    return jsonify(list(segments))


if __name__ == '__main__':
    app.run(debug=True)

```
Please use firefox instead of chrome. With chrome, I also receive the black space. Or you have to go to the network tab to see the API result in chrome.

**Reply by 20May1997** [reply]
Awesome!!! , It worked . Thank you so much
